### DISCUSSION TITLE
har怎么用？

### DISCUSSION THREAD

**Discussion by superen7527** [linweiyuan/go-chatgpt-api#281] [Q&A]
写的脚本自动获取token，但是前几天登录不了了，后来才发现提示需要har。issue只看到一个 https://github.com/linweiyuan/go-chatgpt-api/issues/277，作者说去博客看视频，结果我看不了视频，不知道为啥播放不了挂梯也不行。又找了一下文档也没有文字教程，求助一下：
har应该怎么获取？
获取以后是永久的吗？
获取以后怎么用？可以在post里面加入har吗？api登录有什么改动吗？

好几个月没用了，现在有点不知道怎么解决，之前关注才几个星星没想到就一千stars了。不想打扰发issue就这里吧。希望可以有文字教程或者指路一下 十分感谢。

**Comment by linweiyuan** [comment]
昨天网站挂了，现在视频应该看得了才对

har 不是永久，但是正常用的话，不是很大量请求可以用很久，失效了再获取

获取后参考 compose.yaml 挂载进去，然后正常登录即可不用改东西

**Reply by superen7527** [reply]
好的 我还是看不了视频，不知道什么原因，过几天我再试试吧。叫朋友帮忙看了视频抓了一下har，是这样开头的json文件：
```
{
  "log": {
    "version": "1.2",
    "creator": {
      "name": "WebInspector",
      "version": "537.36"
    },
    "pages": [],
    "entries": [
      {
        "_initiator": {
          "type": "script",
          "stack": {
            "callFrames": [
```


然后根据 https://github.com/linweiyuan/go-chatgpt-api/blob/main/compose.yaml 我把har放root下映射

```
docker run -d --name go-api -p 3300:8080 -e GIN_MODE=release --restart unless-stopped -v /root/chat.openai.com.har:/app/chat.openai.com.har linweiyuan/go-chatgpt-api
```

最后再curl请求post登录还是提示har文件不合法。请问一下是har抓错了，还是我docker映射部署错了？


另外，那个aroke是gpt4用的吗？我看博客视频上面都是说的这个值，gpt3.5能不能用这个，是不是用了可以不需要har？

**Comment by linweiyuan** [comment]
只能本地用 har，服务器没试成功（我不会弄）

3.5 目前不需要这个 arkose token，只有极少数人需要

**Reply by superen7527** [reply]
不太懂你的意思，有几个疑惑：

1.本地指的是本地物理机 window电脑和Linux的docker才能部署，还是指的是本地网络的国内云服务器(实在没明白为什么本地和云服务器的docker有区别)？

2.如果是只能本地部署才能用har，是不是意味着以前那种美国vps直连的登录接口已经不能用来登录了，因为har用不了。只能用本地部署挂上代理的服务才能继续登录？

3.除了har还有别的办法获取token吗？还是说云服务器由于不能用har已经完全不能获取token了？这个相关har有地方讨论吗？我没找到一些像以前会讨论很多楼层的issue，感觉是不是比较冷门。

**Reply by linweiyuan** [reply]
1. 自己电脑可以，服务器不行

2. 对

3. 去官网登录拿，有 tg 群

**Reply by superen7527** [reply]
好的，明白了。也就是现在已经不能全自动登录获取token了，即使har也会过期，希望以后会有新的方法。

**Reply by superen7527** [reply]
不好意思哥再打扰一下：

1.gpt4除了需要 arkose 也需要har吗？如果不需要 我干脆直接找一个4的账号弄全自动登录。

2.这个arkose，看博客的图文，是可以自动获取或者算法计算出来而不需要额外的条件吧？

**Reply by linweiyuan** [reply]
har 就是用来生成 arkose

博客里的方法几个月前，早失效了，现在 har 是最新的方法

**Reply by superen7527** [reply]
好吧 原来如此。
